File: src/main/java/com/lambdaschool/coutrygdp/controller/gdpController.java
package com.lambdaschool.coutrygdp.controller;

import com.lambdaschool.coutrygdp.CoutrygdpApplication;
import com.lambdaschool.coutrygdp.exceptions.NotFound;
import com.lambdaschool.coutrygdp.model.GDP;
//import com.lambdaschool.projectrestdogs.exceptions.ResourceNotFoundException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.ModelAndView;


@RestController
public class gdpController {
    private static final Logger logger = LoggerFactory.getLogger(GDP.class);

    @GetMapping(value = "/gdp")
    public ResponseEntity<?> getAllGDPs()
    {
        logger.info("all the damn gdps your beautiful heart could imagine");
        return new ResponseEntity<>(CoutrygdpApplication.ourGDPList.gdpList, HttpStatus.OK);
    }

    @GetMapping(value = "/gdp/{id}")
    public ResponseEntity<?> getGDPDetail(@PathVariable long id)
    {
        GDP rtnGDP;
        logger.info("you're looking at gdp - " + id + " you filthy animal");

        if ((CoutrygdpApplication.ourGDPList.findGDP(e -> (e.getId()) == id)) == null) {

            throw new NotFound(id + "ain't no gdp silly");

        } else {

            rtnGDP = CoutrygdpApplication.ourGDPList.findGDP(e -> (e.getId() == id));
        }
        return new ResponseEntity<>(rtnGDP, HttpStatus.OK);
    }


    // localhost:2019/dog/table
    @GetMapping(value = "/")
    public ModelAndView displayDogTable()
    {
        ModelAndView mav = new ModelAndView();
        mav.setViewName("gdps");
        mav.addObject("gdpList", CoutrygdpApplication.ourGDPList.gdpList);
        return mav;
    }

}

File: src/main/kotlin/com/jakeesveld/gdp/GdpController.kt
package com.jakeesveld.gdp

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["/gdp"])
class GdpController{

    @GetMapping(value = ["/names"])
    fun getGdpNamesAlphabetized(): ResponseEntity<Any>{
        return ResponseEntity(GdpApplication.gdpList.sortBy { x -> x.name }, HttpStatus.OK)
    }

    @GetMapping(value = ["/economy"])
    fun getGdpByEconomy(): ResponseEntity<Any>{
        return ResponseEntity(GdpApplication.gdpList.sortBy { x -> x.gdp }, HttpStatus.OK)
    }

    @GetMapping(value = ["/country/{id}"])
    fun getGdpById(@PathVariable id: Long):ResponseEntity<Any>{
        return ResponseEntity(GdpApplication.gdpList.filter { x -> x.id == id }, HttpStatus.OK)
    }

    @GetMapping(value = ["/country/stats/median"])
    fun getMedianGdp(): ResponseEntity<Any>{
        val tempList = GdpApplication.gdpList.sortedBy { x -> x.gdp }
        return ResponseEntity(tempList[tempList.size / 2], HttpStatus.OK)
    }



}